public/javascript/controllers/history.js

```diff
@@ -0,0 +1,46 @@
+var viewer = require('../components/viewer');
+var applicationController = require('./application.js');
+
+var initialHref;
+
+function init() {
+    window.addEventListener('popstate', onPopState);
+
+    document.getElementById('viewer').addEventListener('load', function(e){
+        var iframeLocation = e.target.contentWindow.location;
+
+        if (iframeLocation.origin !== "null" || iframeLocation.protocol.indexOf('http') !== -1) {
+            //Needs to be replace (not push) as the iframe has added it's own history entry (shakes fist).
+            replaceLocationHistory(iframeLocation);
```

Comment:
Just had a thought, the iframe location will contain `/proxy` in it after we integrate the proxy into the iframe. This will need to cater for it too.